sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/recover_deleted_key_operation.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A long-running operation for recovering a Key.
+ *
+ */
+
+#pragma once
+
+#include <azure/core/http/http.hpp>
+#include <azure/core/operation.hpp>
+#include <azure/core/operation_status.hpp>
+#include <azure/core/response.hpp>
+
+#include <azure/keyvault/common/internal/keyvault_pipeline.hpp>
+#include <azure/keyvault/common/keyvault_exception.hpp>
+
+#include "azure/keyvault/keys/key_vault_key.hpp"
+
+#include <memory>
+#include <string>
+#include <thread>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+
+  /**
+   * @brief A long running operation to recover a key.
+   *
+   */
+  class RecoverDeletedKeyOperation : public Azure::Core::Operation<KeyVaultKey> {
+  private:
+    /* RecoverDeletedKeyOperation can be constructed only by friends classes (internal creation).
```

Comment:
It also violates [general guidelines](https://azure.github.io/azure-sdk/general_design.html#general-lro-continuation) that LRO pollers must be creatable from some "serialized" state. In .NET, we take the client (in this case, the `KeyClient`) and the key ID as constructor parameters.

Reply:
> This also means the Operation has access to the internals of Operation which we really don't want.

KeyClient (the operation's friend) is a type we own, we can control how it access/consumes/uses the Operation. We expect and audit that it will only ever create the Operation and will never access the internals.  Also, consider that, if we go ahead with the Operation<T> simplification in Azure.Core, the operation won't have any internal/private fields.

Reply:
@heaths you are right. We are breaking that rule in C++.  All of the current Operaion<T> implementations can't be constructed from a previous result of Operation.Poll().  But, since C++ is async only, I don't think if that guideline really applies to C++.